Documentation/cli/substitutepath.md

```diff
@@ -35,32 +36,27 @@ Double quotes can be used to specify paths that contain spaces, or to specify em
 (dlv) config substitute-path /make/this/path/relative ""
 ```
 
+When Delve needs to convert a path P found inside the executable file into a path in the local filesystem it will scan through the list of rules looking for the first one where P starts with the from-path and replace from-path with to-path.
+
+Empty paths in both from-path and to-path are special, they represent relative paths:
+
+- `(from="" to="/home/user/project/src")` converts all relative paths in the executable to absolute paths in `/home/user/project/src`
+- `(from="/build/dir" to="")` converts all paths in the executable that start with `/build/dir` into relative paths.
+
 #### DAP server
 
-If you connect to Delve using the DAP protocol then the substitute path rules are specified using the substitutePath option in [launch.json](https://github.com/golang/vscode-go/blob/master/docs/debugging.md#launchjson-attributes).
+If you connect to Delve using the DAP protocol then the substitute path rules are specified using the substitutePath option in [launch.json](https://github.com/golang/vscode-go/blob/master/docs/debugging.md#launchjson-attributes). **Note that `from` and `to` have reversed meaning here.**
 
 ```
 	"substitutePath": [
-		{ "from": "/from/path", "to": "/to/path" }
+		{ "from": "/debugger/machine/directory", "to": "/compiler/machine/directory" }
 	]
 ```
 
 The [debug console](https://github.com/golang/vscode-go/blob/master/docs/debugging.md#dlv-command-from-debug-console) can also be used to modify the path substitution list:
 
 ```
-dlv config substitutePath /from/path /to/path
+dlv config substitutePath /debugger/machine/directory /compiler/machine/directory
 ```
 
-This command works similarly to the `config substitute-path` command described above.
-
-### How are path substitution rules applied
-
-Regardless of how they are specified the path substitution rules are an ordered list of `(from-path, to-path)` pairs. When Delve needs to convert a path P found inside the executable file into a path in the local filesystem it will scan through the list of rules looking for the first one where P starts with from-path and replace from-path with to-path.
-
-Empty paths in both from-path and to-path are special, they represent relative paths: 
-
-- `(from="" to="/home/user/project/src")` converts all relative paths in the executable to absolute paths in `/home/user/project/src`
-- `(from="/build/dir" to="")` converts all paths in the executable that start with `/build/dir` into relative paths.
-
-The path substitution code is SubstitutePath in pkg/locspec/locations.go.
-
+When Delve needs to convert a path P found in the local filesystem into a path inside the executable file it will scan through the list of rules looking for the first one where P starts with from-path and replace from-path with to-path.
```

Comment:
This should be move above, with the rest of the paragraph.

Reply:
I have moved the paragraph up and re-introduced the section about the special meaning of empty paths. I have removed the related section from the commit message.

For consistency, I have also moved the similar paragraph in the cli section up.